File: easy.server.demo/easy.server.netty.demo/src/main/java/io/github/jinlongliao/easy/server/netty/demo/logic/annotation/UserId.java
package io.github.jinlongliao.easy.server.netty.demo.logic.annotation;

import io.github.jinlongliao.easy.server.core.annotation.LogicAlias;
import io.github.jinlongliao.easy.server.core.annotation.LogicRequestParam;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author: liaojinlong
 * @date: 2022-09-16 10:00
 */
@LogicRequestParam(value = "userId", length = 4, defaultValue = "234")
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface UserId {
    /**
     * 参数名称
     *
     * @return 参数名称
     */
    @LogicAlias("value")
    String newV() default "userId";
}

File: easy.server.demo/easy.server.kotlin.demo/src/main/kotlin/io/github/jinlongliao/easy/server/kotlin/demo/logic/annotation/UserId.kt
package io.github.jinlongliao.easy.server.kotlin.demo.logic.annotation

import io.github.jinlongliao.easy.server.core.annotation.LogicAlias
import io.github.jinlongliao.easy.server.core.annotation.LogicRequestParam

/**
 * @author: liaojinlong
 * @date: 2022-09-16 10:00
 */
@LogicRequestParam(value = "userId", length = 4, defaultValue = "234")
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class UserId(
    /**
     * 参数名称
     *
     * @return 参数名称
     */
    @get:LogicAlias("value") val newV: String = "userId"
)
